pam/authorization.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/internal/log"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// sendIsAuthorized sends the authorization challenges or wait request to the brokers.
+// The event will contain the returned value from the broker.
+func sendIsAuthorized(ctx context.Context, client authd.PAMClient, sessionID, content string) tea.Cmd {
+	return func() tea.Msg {
+		res, err := client.IsAuthorized(ctx, &authd.IARequest{
+			SessionId:          sessionID,
+			AuthenticationData: content,
+		})
+		if err != nil {
+			if st := status.Convert(err); st.Code() == codes.Canceled {
+				return isAuthorizedResultReceived{
+					access: "cancelled",
```

Comment:
We can’t here, because as we cancel the context, there is no access to the "access" field (the struct is nil). However, I agree, we can resend responses.AuthCancelled.
Will change that.